core/server/master/src/main/java/alluxio/master/file/InodeTtlChecker.java

```diff
@@ -131,12 +139,20 @@ public void heartbeat() throws InterruptedException {
                 LOG.error("Unknown ttl action {}", ttlAction);
             }
           } catch (Exception e) {
-            LOG.error("Exception trying to clean up {} for ttl check", inode, e);
+            LOG.error("Exception trying to clean up {} for ttl check", path, e);
+            if (inode != null) {
+              failedInodes.add(inode);
```

Comment:
What cases could there be an exception here?
Is it possible that there could be a retry here forever? e.g. maybe if ACL is changed?

Reply:
so if there's any exception like file not exist or any file status changes, it would be again loaded from inode store and check ttl again to ensure if it actually needs a retry. (if exception like filenotexist in first retry, the next retry will just be dropping this inode from ttlbucket record) what do u mean when ACL is changed? in this case where the delete op is issued from master within, what user will it be permission checked against?

Reply:
added the retry attempt to address this concern, so we won't retry indefinitely. thanks